coordinator/src/admin.rs

```diff
@@ -144,6 +156,98 @@ pub async fn list_dlc_channels(
     Ok(Json(dlc_channels))
 }
 
+pub async fn collaborative_revert(
+    State(state): State<Arc<AppState>>,
+    revert_params: Json<CollaborativeRevert>,
+) -> Result<Json<String>, AppError> {
+    let channel_id_string = revert_params.channel_id.clone();
+    let channel_id = hex::decode(channel_id_string.clone())
+        .map_err(|err| AppError::BadRequest(err.to_string()))?;
+    let channel_id: [u8; 32] = channel_id
+        .try_into()
+        .map_err(|_| AppError::BadRequest("Provided channel ID was invalid".to_string()))?;
+
+    let mut conn =
+        state.pool.clone().get().map_err(|e| {
+            AppError::InternalServerError(format!("Failed to acquire db lock: {e:#}"))
+        })?;
+
+    let channel_details = state
+        .node
+        .inner
+        .channel_manager
+        .get_channel_details(&channel_id)
+        .context("Could not get channel")
+        .map_err(|error| {
+            AppError::InternalServerError(format!("No ln channel found: {error:#}"))
+        })?;
+
+    let sub_channels = state.node.inner.list_dlc_channels().map_err(|e| {
+        AppError::InternalServerError(format!("Failed to list DLC channels: {e:#}"))
+    })?;
+    let subchannel = sub_channels
+        .iter()
+        .find(|c| c.channel_id == channel_id)
+        .context("Could not find provided channel")
+        .map_err(|e| AppError::BadRequest(format!("Channel not found: {e:#}")))?;
+
+    let position = Position::get_position_by_channel_id(&mut conn, channel_id_string.clone())
+        .map_err(|error| {
+            tracing::error!(
+                channel_id = revert_params.channel_id.clone(),
+                "Could not get position for channel {error:#}"
+            );
+            AppError::InternalServerError(format!("Failed to load position from db: {error:#}"))
+        })?;
+
+    let settlement_amount = position
```

Comment:
why not use the `unrealized_pnl_sat` from the position?

Reply:
ok, looks like the `unrealized_pnl_sat` is never set 🙈

Reply:
😅 I didn't trust these values. Hence I didn't go for it